content/docs/tech-insights/track-deprecated-actions/index.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Tracking Node.js Deprecated Actions and Commands
+publishedDate: '2024-03-08T15:15:00.0Z'
+description: How to track and visualize the deprecation notices from GitHub related to your node.js software with Roadie Tech Insights
+---
+
+Tracking deprecated actions and commands is critical for ensuring your software is secure and compliant. It can be difficult to track which of your software is using a supported action, and which are using an deprecated.
+
+With Roadie’s Tech Insights feature, there’s an easier way. In this tutorial, we’re going to:
+
+1. Automatically scan Components in the Backstage catalog and record which components have been identified by GitHub as using a deprecated action or command.
+2. Visualise the distribution of deprecated actions or commands.
+3. Create a check that can tell us which software is using deprecated actions or commands.
+
+As we go through this process, you’ll learn:
+
+1. How to create Data Sources, Checks and Scorecards with Tech Insights.
+2. How to parse files with JSONata in Roadie and extract facts from those GraphQL queries.
+3. How to detect which repositories have GitHub deprecation notices with Tech Insights.
+4. How to use the “Does not contain RegExp” operator to do comparisons.
+
+Let’s get started.
+
+## Record the deprecation notices automatically
+
+In order to track the deprecation notices from GitHub, we are going to create a Data Source In Roadie Tech Insights. It will be setup to run on a scheduled basis and extract the notices.
+
+### Data Source
+
+1. Visit Tech Insights, select the **Data Sources** tab, and press **ADD DATA SOURCE**.
+
+   ![a button on a web interface](./Add_Data_Source.png)
+
+2. Enter a descriptive **Name** and **Description**.
+
+   ![About section input fields](./Data_About_Section.png)
+
+3. Set the Data Provider **Type** to `HTTP`, **Proxy** to `/github/api`, **Path Extension** to `graphql`, **HTTP Method** to POST, and **Body** to the below. 
```

Comment:
Nothing to do for this, but we will expose the POST endpoints for `Type` `GitHub API` also, which allows the users to use the credentials from installed GitHub app instead of some token they have stored in as a secret